kotlin-eclipse-core/src/org/jetbrains/kotlin/core/utils/ProjectUtils.kt

```diff
@@ -0,0 +1,331 @@
+import org.jetbrains.kotlin.core.KotlinClasspathContainer
+import org.jetbrains.kotlin.core.builder.KotlinPsiManager
+import org.jetbrains.kotlin.core.log.KotlinLogger
+import org.jetbrains.kotlin.core.model.KotlinNature
+import org.jetbrains.kotlin.psi.KtFile
+import java.io.File
+import java.io.IOException
+import java.util.*
+
+object ProjectUtils {
+
+    private const val LIB_FOLDER = "lib"
+    private const val LIB_EXTENSION = "jar"
+
+    private const val MAVEN_NATURE_ID = "org.eclipse.m2e.core.maven2Nature"
+    private const val GRADLE_NATURE_ID = "org.eclipse.buildship.core.gradleprojectnature"
+
+    @JvmStatic
+    val KT_HOME = ktHome
+
+    val accessibleKotlinProjects: List<IProject>
+        get() = ResourcesPlugin.getWorkspace().root.projects.filter { project -> isAccessibleKotlinProject(project) }
+
+    private val ktHome: String
+        get() = try {
+            val compilerBundle = Platform.getBundle("org.jetbrains.kotlin.bundled-compiler")
+            FileLocator.toFileURL(compilerBundle.getEntry("/")).file
+        } catch (e: IOException) {
+            KotlinLogger.logAndThrow(e)
+        }
+
+    fun getJavaProjectFromCollection(files: Collection<IFile>): IJavaProject? {
+        var javaProject: IJavaProject? = null
+        for (file in files) {
+            javaProject = JavaCore.create(file.project)
+            break
+        }
+
+        return javaProject
+    }
+
+    fun getPackageByFile(file: IFile): String? {
+        val jetFile = KotlinPsiManager.getParsedFile(file)
+
+        return jetFile.packageFqName.asString()
+    }
+
+	@JvmStatic
+    @JvmOverloads
+    fun cleanFolder(container: IContainer?, predicate: (IResource) -> Boolean = { true }) {
+        try {
+            if (container == null) {
+                return
+            }
+            if (container.exists()) {
+                for (member in container.members()) {
+                    if (member is IContainer) {
+                        cleanFolder(member, predicate)
+                    }
+                    if (predicate(member)) {
+                        if (member.exists()) {
+                            member.delete(true, null)
+                        }
+                    }
+                }
+            }
+        } catch (e: CoreException) {
+            KotlinLogger.logError("Error while cleaning folder", e)
+        }
+
+    }
+
+	@JvmStatic
+    fun getOutputFolder(javaProject: IJavaProject): IFolder? = try {
+        ResourcesPlugin.getWorkspace().root.findMember(javaProject.outputLocation) as IFolder
+    } catch (e: JavaModelException) {
+        KotlinLogger.logAndThrow(e)
+    }
+
+
+    fun getSourceFiles(project: IProject): List<KtFile> {
+        val jetFiles = ArrayList<KtFile>()
+        for (file in KotlinPsiManager.getFilesByProject(project)) {
+            val jetFile = KotlinPsiManager.getParsedFile(file)
+            jetFiles.add(jetFile)
+        }
+
+        return jetFiles
```

Comment:
`KotlinPsiManager.getFilesByProject(project).map { KotlinPsiManager.getParsedFile(it) }`